### PR TITLE
Added validation for required tag name

### LINKED ISSUE #131: 500 error when content template has no RLP tag
If the `sulu.rlp` tag is required then there should be some kind of validation - or the URL should be automatically generated if this tag is absent based on the node-name.

### PR DESCRIPTION
**Pull request by wachterjohannes**
__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes
- [x] added validation in template reader
- [x] adapted testcases

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | fixes #131 
| BC breaks        | https://github.com/sulu-io/sulu/commit/96f156409bdca0f586f7cd6ce6d9f788b74aa5d0 
| Documentation PR | none

### CODE REVIEW

**Comment by wachterjohannes** [comment]
note for me: tests fail because of missing rlp tags in testcases!

**Comment by wachterjohannes** [comment]
@danrot i found the problem in the tests! we have an `excerpt.xml` which is interpreted as an page template but has no `rlp-tag` ... therefor we should handle them not as pages ... which will be a "bigger" task ...

File: src/Sulu/Component/Content/Template/TemplateReader.php [hunk 1]
```diff
@@ -101,6 +109,15 @@ public function load($resource, $type = Structure::TYPE_PAGE)
             }
         }
 
+        // FIXME until excerpt-template is no page template anymore
```
Comment (danrot) [inline src/Sulu/Component/Content/Template/TemplateReader.php]: Can you somehow link this to our issue for that?

File: tests/Resources/DataFixtures/Page/template_duplicated_priority.xml [hunk 2]
```diff
@@ -29,5 +29,9 @@
                 <param name="maxLinks" value="10"/>
             </params>
         </property>
+
+        <property name="url" type="text_line">
+            <tag name="sulu.rlp" />
```
Comment (danrot) [inline tests/Resources/DataFixtures/Page/template_duplicated_priority.xml]: Is it possible to have multiple `sulu.rlp`-tags per template?

**Comment by danrot** [comment]
How did you solve your last comment now? Did you do this refactoring you have talked about?

**Comment by dantleech** [comment]
Note that this has been replaced in the refactoring branch.

**Comment by dantleech** [comment]
The "TemplateReader" has been moved to `LegacyXmlLoader`, from which the `XmlLoader` extends. I am not sure how much of a pain this will be to merge.

File: tests/Resources/DataFixtures/Page/template_duplicated_priority.xml (hunk 2 above)
Comment (wachterjohannes) [inline tests/Resources/DataFixtures/Page/template_duplicated_priority.xml]: yes if you do this the one with the highest priority will be used!

**Comment by wachterjohannes** [comment]
@danrot i would do this refactoring in a new PR perhaps for 1.1 ...

**Comment by danrot** [comment]
What would this refactoring be about?

**Comment by wachterjohannes** [comment]
https://github.com/sulu-io/sulu/issues/1220#issuecomment-110704259

**Comment by wachterjohannes** [comment]
it is linked with that `internal` templates 

* excerpt
* internal_link
* external_link

they should not be a XML-template but at least excerpt should be able to overwriten

File: src/Sulu/Component/Content/Template/TemplateReader.php (hunk 1 above)
Comment (wachterjohannes) [inline src/Sulu/Component/Content/Template/TemplateReader.php]: fixed

**Comment by danrot** [comment]
@dantleech Is this change also ok for you?

**Comment by chirimoya** [comment]
ping ... @danrot @dantleech @wachterjohannes

**Comment by wachterjohannes** [comment]
for me its finished (=

**Comment by danrot** [comment]
@wachterjohannes Why are all the `sulu.rlp.part` tags replaced by `sulu.rlp`? I hope both tags are still existing?

@dantleech How does this affect the refactoring?

**Comment by wachterjohannes** [comment]
`sulu.rlp.part` is only used in the frontend ... its useless in the testcases and i need there the `sulu.rlp` tag! so i have replaced it!

File: src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php [hunk 3]
```diff
@@ -59,7 +70,7 @@ class XmlLegacyLoader implements LoaderInterface
     /**
      * {@inheritdoc}
      */
-    public function load($resource, $type = 'page')
+    public function load($resource, $type = 'page', $internal = false)
```
Comment (wachterjohannes) [inline src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php]: @danrot @dantleech did we have no constants for that types?

Comment (danrot) [inline src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php]: I can't remember any.

Comment (wachterjohannes) [inline src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php]: before it was on the `Structure` class

**Comment by danrot** [comment]
@wachterjohannes Tests are failing.

File: UPGRADE.md [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 ## dev-develop
 
+<<<<<<< HEAD
```
Comment (danrot) [inline UPGRADE.md]: Shouldn't be here :-)

**Pull request merged**